### PR TITLE
fix: add support for arrays in the application.properties parser.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: core/src/main/java/io/ap4k/utils/Maps.java [hunk 1]
```diff
@@ -106,11 +111,59 @@
...
-    Object nested = asMap(remaining, value);
+    Map<String, Object> nested = asMap(remaining, value);
     result.put(key, nested);
     return result;
   }
 
+  /**
+   * Convert a multipart-key value pair to a Map.
+   */
+  private static Map<String, Object> asMapWIP(String[] keys, Object value) {
```
Comment (geoand) [inline core/src/main/java/io/ap4k/utils/Maps.java]: what's with the name? :P

File: readme.md [hunk 2]
```diff
@@ -865,8 +865,80 @@ By adding the annotation to your test class the following things will happen:
...
+      kubernetes:
+        labels:
+          - key: foo
+            value: bar
+   
+   
+In the examples above, `ap4k` is the prefix that we use to `namespace` the ap4k configuration. `kubernetes` defines the annotation kind (its @KubernetesApplication in lower case and stripped of the `Application` suffix).
+`labels`, `key` and value are the property names and since the `Label` is nested under `@KubernetesApplication` so are the properties.
+
+The exact same example for openshift (where `@OpneshiftApplication` is used instead) would be:
```
Comment (geoand) [inline readme.md]: `OpneshiftApplication` -> `OpenshiftApplication`

File: readme.md [hunk 3]
```diff
@@ -865,8 +865,80 @@ By adding the annotation to your test class the following things will happen:
...
+or using yaml:
+
+    ap4k:
+      kubernetes:
+        labels:
+          - key: foo
+            value: bar
+   
+   
+In the examples above, `ap4k` is the prefix that we use to `namespace` the ap4k configuration. `kubernetes` defines the annotation kind (its @KubernetesApplication in lower case and stripped of the `Application` suffix).
```
Comment (geoand) [inline readme.md]: missing \` around `@KubernetesApplication`

File: core/src/main/java/io/ap4k/utils/Maps.java [hunk 4]
```diff
@@ -21,15 +21,18 @@
...
 import java.util.regex.Pattern;
 
 public class Maps {
   private static final String PROPERTY_PREFIX = "ap4k";
   private static final String MULTIPART_SEPARATOR_PATTERN = Pattern.quote(".");
-
+  private static final String ARRAY_PATTERN = "[a-zA-Z0-9_]+\\[\\d+\\]";
 
   /**
    * Read a properties input stream and crate a configuration map.
```
Comment (aureamunoz) [inline core/src/main/java/io/ap4k/utils/Maps.java]: create ?

File: core/src/main/java/io/ap4k/utils/Maps.java (hunk 1 above)
Comment (iocanel) [reply core/src/main/java/io/ap4k/utils/Maps.java]: meant to be removed.

Comment (geoand) [reply core/src/main/java/io/ap4k/utils/Maps.java]: Kill it then :P

**Pull request merged**